### ISSUE TITLE
should not throw error at `.d.ts` when `func` + `namespace` has member `default`

### ISSUE THREAD

**Issue by bluelovers** [microsoft/TypeScript#32470]
**TypeScript Version:**  3.4.0-dev.201xxxxx

**Search Terms:**

**Code**

.d.ts
```ts
declare function execAll<T extends RegExp = RegExp>(inputRegExp: T | RegExp, input: string, options?: IExecAllOptions<T>): IMatches<T>;
declare namespace execAll {
    var default: typeof execAll;
}
import IExecAllOptions = execAll.IExecAllOptions;
import IMatches = execAll.IMatches;
declare namespace execAll {
    function execall<T extends RegExp = RegExp>(inputRegExp: T | RegExp, input: string, options?: IExecAllOptions<T>): IMatches<T>;
    interface IExecAllOptions<T extends RegExp = RegExp> {
        resetLastIndex?: boolean;
        /**
         * allow change cloneRegexp function
         */
        cloneRegexp?: ICloneRegexp<T>;
        /**
         * only use this when u know what u doing
         */
        leftContext?: boolean;
        rightContext?: boolean;
        removeHiddenData?: boolean;
    }
    interface ICloneRegexp<T extends RegExp = RegExp> {
        (inputRegExp: T | RegExp, ...argv: any[]): T;
    }
    type IExecAllRegExpExecArray<T extends RegExp = RegExp> = RegExpExecArray & string[] & {
        /**
         * The 0-based index of the match in the string.
         */
        index: number;
        /**
         * es2018
         */
        groups?: {
            [k: string]: string;
        };
    };
    type IMatches<T extends RegExp = RegExp> = (IExecAllRegExpExecArray<T> & {
        match: string;
        sub: string[];
        leftContext?: string;
        rightContext?: string;
    })[] & {
        /**
         * regular expressions
         *
         * @readonly
         */
        readonly re: T;
        /**
         * regular expressions that contains the string against which a regular expression is matched.
         *
         * @readonly
         */
        readonly input: string;
        /**
         * last matched index
         *
         * @readonly
         */
        readonly lastIndex: number;
    };
    const SYMBOL: unique symbol;
}
export = execAll;

```

**Expected behavior:**

> no error

**Actual behavior:**

![image](https://user-images.githubusercontent.com/167966/61464421-cd91e000-a9a8-11e9-8ec6-c00623d18f43.png)

```
Error:(3, 9) TS1134: Variable declaration expected.
Error:(3, 16) TS1134: Variable declaration expected.
Error:(3, 18) TS1134: Variable declaration expected.
```

**Playground Link:** 

**Related Issues:**

**Comment by AnyhowStep** [comment]
This?
```ts
declare namespace execAll {
    var blah : number;
    export {
        blah as default
    }   
}
execAll.default
```

**Comment by bluelovers** [comment]
this code
```ts
declare function execAll<T extends RegExp = RegExp>(inputRegExp: T | RegExp, input: string, options?: IExecAllOptions<T>): IMatches<T>;
declare namespace execAll {
    var default: typeof execAll;
}
```
is made from
```ts
execAll.default = execAll;
```

**Comment by RyanCavanaugh** [comment]
Fixed in TypeScript 4.0.0-dev.20200611 by #38982 (`08cb0b2`).

With `--declaration --emitDeclarationOnly --module commonjs`, TypeScript 3.9.10 emitted the function expando `execAll.default = execAll` as:

```ts
declare namespace execAll {
    var default: typeof execAll;
}
```

Compiling that emitted declaration reports TS1134. Starting with 4.0.0-dev.20200611, it instead emits a private alias and `export { _a as default }`, which compiles successfully. Current TypeScript 7.1.0-dev.20260902.1 emits a valid `export { execAll as default }` declaration, and that declaration can be consumed successfully.
